src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -443,6 +465,57 @@ private static String getNodeClassNameWithoutContext(ParseTree node) {
     }
 
     /**
+     * Method to search for a missed HTML tag which could possibly be responsibe for ANTLR error.
+     * @param exception {@code NoViableAltException} object catched while parsing javadoc
+     * @return returns appropriate {@code ParseErrorMessage} if a HTML close tag is missed;
+     *     null otherwise
+     */
+    private ParseErrorMessage generateMissedHtmlTagErrorMessage(RecognitionException exception) {
+        ParseErrorMessage parseErrorMessage = null;
+        final Interval sourceInterval = exception.getCtx()
+                .getSourceInterval();
+        final List<Token> tokenList;
+        if (sourceInterval.length() == 0) {
+            tokenList = new ArrayList<>(0);
+        }
+        else {
+            tokenList = ((BufferedTokenStream) exception.getInputStream())
+                    .getTokens(sourceInterval.a, sourceInterval.b);
+        }
+        final ArrayDeque<Token> stack = new ArrayDeque<>();
+        final ListIterator<Token> tokenListIterator = tokenList.listIterator();
+        Token token;
+        Token htmlTagNameStart;
+        while (tokenListIterator.hasNext()) {
+            token = tokenListIterator.next();
+            if (token.getType() == JavadocTokenTypes.HTML_TAG_NAME
+                    && tokenList.get(tokenListIterator.previousIndex() - 1)
+                    .getType() != JavadocTokenTypes.SLASH) {
+                stack.push(token);
+            }
+            else if (token.getType() == JavadocTokenTypes.HTML_TAG_NAME) {
```

Comment:
This `else if` branch is to be only taken if the one above it wasn't take for sure. To extract it separately would be to increase cyclomatic complexity by 1 as I would have to add one more condition to the extracted branch.